src/core/renderer.js

```diff
@@ -186,7 +182,14 @@ class Renderer {
                     continue;
                 }
 
-                viewportExpressions.forEach(viewportExpression => viewportExpression.accumViewportAgg(feature));
+                viewportExpressions.forEach(viewportExpression => {
+                    if (viewportExpression._requiredProperties) {
+                        const propNames = viewportExpression._requiredProperties.map(p => p.name);
```

Comment:
I think this should go into the viewport agg instead of here, both for the filtering of properties and for the decoding of categories

Reply:
Agreed, specially since you're modifying the renderer to cache the features per dataframe in another PR. The problem I see is how to get the metadata in an expression that won't be compiled. I can do without the metadata to get the name of the required properties by using their MNS, but I need also the `IDToCategory` to decode categories... I'll need some help to decide how to do that.